### PR TITLE
feat(clips): add clips to the beam client

### PR DESCRIPTION
**Pull request by chrwhit**
Add clipping to the beam client.

Please Look at:
A few of the endpoints do not return objects, just the fact that they return 200 is sufficient. For these enpoints I removed the IResponse portion and just had them return true or false. It looks kind of weird because it breaks the pattern. Thought? Also, if we want to go back to IResponse, what is the <T> for no body?

### CODE REVIEW

File: src/services/Clips.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+export class ClipsService extends Service {
+
+    /**
+     * Can a clip be created on the given broadcast.
+     */
+    public async canClip(broadcastId: string): Promise<boolean> {
+        try {
+            await this.makeHandled<boolean>('get', `clips/broadcasts/${broadcastId}/canClip`);
+        } catch {
+            return Promise.resolve(false);
```
Comment (ProbablePrime) [inline src/services/Clips.ts]: Why can't I clip? Does the API provide this?

File: src/services/Clips.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    /**
+     * Creates a clip.
+     */
+    public async createClip(p: IClipRequest): Promise<boolean> {
+        try {
+            await this.makeHandled<boolean>('post', `clips/create`, { body: p });
+        } catch {
+            return Promise.resolve(false);
```
Comment (ProbablePrime) [inline src/services/Clips.ts]: Why couldn't the clip be created? Does the api provide this?

File: src/services/Clips.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    /**
+     * Deletes a clip.
+     */
+    public async deleteClip(shareableId: string): Promise<boolean> {
+        try {
+            await this.makeHandled<boolean>('delete', `clips/delete/${shareableId}`);
+        } catch {
+            return Promise.resolve(false);
```
Comment (ProbablePrime) [inline src/services/Clips.ts]: Why can't I delete a clip? Does the API provide this?

File: src/services/Clips.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    /**
+     * Renames a clip.
+     */
+    public async renameClip(shareableId: string, newTitle: string): Promise<boolean> {
+        try {
+            await this.makeHandled<IClipProperties>('post', `clips/${shareableId}/metadata`, { body: { title: newTitle } });
+        } catch {
+            return Promise.resolve(false);
```
Comment (ProbablePrime) [inline src/services/Clips.ts]: Why can't I rename the clip?

File: src/services/Clips.ts [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        }
```
Comment (connor4312) [inline src/services/Clips.ts]: Swallowing all these errors is no bueno, as Richard mentioned. Even if there's only one 'expected' cause for failure today, there's always network issues or 5xx issues, and there may be more intended failure causes in the future. Clients will need to be able to to differentiate them.

File: src/defs/clipProperties.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
+import { ILocator } from './locator';
+
+export interface IClipProperties {
+    //The content ID of this clip.
```
Comment (SimonSchick) [inline src/defs/clipProperties.ts]: Please use doc tags eg.
```
/**
 * The content ID of this clip.
 */
```

File: src/services/Clips.ts [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+     * Can a clip be created on the given broadcast.
+     */
+    public async canClip(broadcastId: string): Promise<boolean> {
+        try {
+            await this.makeHandled<boolean>('get', `clips/broadcasts/${broadcastId}/canClip`);
+        } catch {
+            return Promise.resolve(false);
+        }
+
+        return Promise.resolve(true);
```
Comment (SimonSchick) [inline src/services/Clips.ts]: Inside async functions you do not need to return promises, everything returned will be automatically wrapped.
Simply `return true;` here.

File: src/services/Clips.ts [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    /**
+     * Creates a clip.
+     */
+    public async createClip(p: IClipRequest): Promise<boolean> {
```
Comment (SimonSchick) [inline src/services/Clips.ts]: Please avoid single char variable names.

Comment (chrwhit) [reply src/services/Clips.ts]: Opps. Meant to change this

File: src/services/Clips.ts [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+     */
+    public getClip(shareableId: string): Promise<IResponse<IClipProperties>> {
+        return this.makeHandled<IClipProperties>('get', `clips/${shareableId}`);
+    }
+
+    /**
+     * Renames a clip.
+     */
+    public async renameClip(shareableId: string, newTitle: string): Promise<IResponse<IClipProperties>> {
+        return this.makeHandled<IClipProperties>('post', `clips/${shareableId}/metadata`, { body: { title: newTitle } });
```
Comment (connor4312) [inline src/services/Clips.ts]: This doesn't match what's in the API docs -- the docs say the body should be a plain string

**Pull request merged**